pyth_observer/events.py

```diff
@@ -147,7 +147,9 @@ def is_valid(self) -> bool:
         # The normalized confidence
         self.confidence = abs(delta / self.publisher_aggregate.confidence_interval)
 
-        if self.confidence > self.threshold:
+        if (self.price.is_publishing(self.publisher_key) and
+                self.price.is_aggregate_publishing() and
+                self.confidence > self.threshold):
```

Comment:
It's totally fine to do this here, but it means this code: https://github.com/pyth-network/pyth-observer/blob/main/pyth_observer/prices.py#L66-L83

in verify_price(): https://github.com/pyth-network/pyth-observer/blob/main/pyth_observer/prices.py#L156

Is not working properly for some reason. The logic came from [this code](https://github.com/pyth-network/publisher-utils/blob/main/src/validation.ts#L136-L143) in publisher-utils from @jayantk.